src/Neo/Ledger/MemoryPool.cs

```diff
@@ -320,6 +341,44 @@ internal VerifyResult TryAdd(Transaction tx, DataCache snapshot)
             return VerifyResult.Succeed;
         }
 
+        /// <summary>
+        /// Checks whether transaction conflicts with mempooled verified transactions
+        /// and can be added to the pool. If true, then transactions from conflictsList
+        /// should be removed from the verified structures (_unsortedTransactions and
+        /// _sortedTransactions).
+        /// </summary>
+        /// <param name="tx">The <see cref="Transaction"/> that needs to be checked.</param>
+        /// <param name="conflictsList">The list of conflicting verified transactions that should be removed from the pool if tx fits the pool.</param>
+        /// <returns>True if transaction fits the pool, otherwise false.</returns>
+        private bool CheckConflicts(Transaction tx, out List<PoolItem> conflictsList)
+        {
+            conflictsList = new();
+            // Step 1: check if `tx` was in attributes of mempooled transactions.
+            if (_conflicts.TryGetValue(tx.Hash, out var conlicting))
+            {
+                foreach (var hash in conlicting)
+                {
+                    var existingTx = _unsortedTransactions[hash];
+                    if (existingTx.Tx.Signers.Select(s => s.Account).Contains(tx.Sender) && existingTx.Tx.NetworkFee > tx.NetworkFee) return false;
+                    conflictsList.Add(existingTx);
+                }
+            }
+            // Step 2: check if mempooled transactions were in `tx`'s attributes.
+            foreach (var hash in tx.GetAttributes<Conflicts>().Select(p => p.Hash))
+            {
+                if (_unsortedTransactions.TryGetValue(hash, out PoolItem existingTx))
+                {
+                    if (!tx.Signers.Select(p => p.Account).Contains(existingTx.Tx.Sender)) return false;
+                    if (existingTx.Tx.NetworkFee >= tx.NetworkFee) return false;
+                    conflictsList.Add(existingTx);
+                }
+            }
+            // Step 3: take into account sender's conflicting transactions while balance check,
+            // this will be done in VerifyStateDependant.
+
+            return true;
+        }
```

Comment:
```suggestion
        /// <summary>
        /// Checks whether there is no mismatch in conflict attributes between current transaction and mempooled unsorted transactions.
        /// If true, then these unsorted transactions will be added into conflictsList.
        /// </summary>
        /// <param name="tx">The <see cref="Transaction"/>current transactions needs to be checked.</param>
        /// <param name="conflictsList">The list of conflicting verified transactions that should be removed from the pool if tx fits the pool.</param>
        /// <returns>True if transaction fits the pool, otherwise false.</returns>
        private bool CheckConflicts(Transaction tx, out List<PoolItem> conflictsList)
        {
            conflictsList = new();
            // Step 1: check if `tx` was in conflict attributes of unsorted transactions.
            if (_conflicts.TryGetValue(tx.Hash, out var conlicting))
            {
                foreach (var hash in conlicting)
                {
                    var unsortedTx = _unsortedTransactions[hash];
                    if (unsortedTx.Tx.Signers.Select(s => s.Account).Contains(tx.Sender) && unsortedTx.Tx.NetworkFee > tx.NetworkFee) return false;
                    conflictsList.Add(unsortedTx);
                }
            }
            // Step 2: check if unsorted transactions were in `tx`'s conflict attributes.
            foreach (var hash in tx.GetAttributes<Conflicts>().Select(p => p.Hash))
            {
                if (_unsortedTransactions.TryGetValue(hash, out PoolItem unsortedTx))
                {
                    if (!tx.Signers.Select(p => p.Account).Contains(unsortedTx.Tx.Sender)) return false;
                    if (unsortedTx.Tx.NetworkFee >= tx.NetworkFee) return false;
                    conflictsList.Add(unsortedTx);
                }
            }
            // Step 3: take into account sender's conflicting transactions while balance check,
            // this will be done in VerifyStateDependant.

            return true;
        }
```
Maybe this way is more clear.

Reply:
> I've fixed this in a separate commit and fixed some typos along the way. Could you review one more time, please?

Now it's clear enough! Excellent!